source/fe/mapping_cartesian.cc

```diff
@@ -667,6 +667,21 @@ MappingCartesian<dim,spacedim>::transform (
         for (unsigned int d1=0; d1<dim; ++d1)
           for (unsigned int d2=0; d2<dim; ++d2)
             output[i][d1][d2] = input[i][d1][d2] * data.length[d2]
+                                / data.volume_element;
+      return;
+    }
```

Comment:
Are you saying that the `return` statement was simply missing here and that we are accidentally following through into the next `case` block? That looks quite plausible, looking at the code around it.